sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -180,9 +195,26 @@ export class Receiver {
     cancellationToken?: Aborter
   ): Promise<ReceivedEventData[]> {
     this._throwIfReceiverOrConnectionClosed();
+    this._throwIfAlreadyReceiving();
+    let lastStreamingReceiverSequenceNum: number | undefined;
+    if (this._streamingReceiver) {
+      lastStreamingReceiverSequenceNum = this._streamingReceiver.getSequenceNumber();
+    }
+
     if (!this._batchingReceiver) {
+      if (lastStreamingReceiverSequenceNum && lastStreamingReceiverSequenceNum > -1 && this._receiverOptions) {
+        this._receiverOptions.eventPosition = EventPosition.fromSequenceNumber(lastStreamingReceiverSequenceNum);
+      }
       this._batchingReceiver = BatchingReceiver.create(this._context, this.partitionId, this._receiverOptions);
+    } else {
+      const lastBatchingReceiverSequenceNumber = this._batchingReceiver.getSequenceNumber();
+      if (lastStreamingReceiverSequenceNum && lastStreamingReceiverSequenceNum > lastBatchingReceiverSequenceNumber) {
+        await this._batchingReceiver.close();
+        this._receiverOptions.eventPosition = EventPosition.fromSequenceNumber(lastStreamingReceiverSequenceNum);
+        this._batchingReceiver = BatchingReceiver.create(this._context, this.partitionId, this._receiverOptions);
```

Comment:
I believe a lot of this logic will be used inside `receive()` as well. What do you think of having a `getCheckpoint()` on the `Receiver` class which will encapsulate all this logic in it? i.e. it will give the last received event's sequence number on this receiver.

The `receive()` function will take this checkpoint and start a new streaming receiver.
The `receiveBatch()` function will compare this with the value of the checkpoint on the batching receiver and take a call if it needs to create a new batching receiver or not